server/src/service.rs

```diff
@@ -123,46 +125,85 @@ impl Service {
 }
 
 impl Service {
-    pub async fn run() -> Result<()> {
+    pub async fn run() -> Result<zbus::Connection> {
         let connection = zbus::connection::Builder::session()?
             .name(oo7::dbus::api::Service::DESTINATION.as_deref().unwrap())?
             .build()
             .await?;
-        let object_server = connection.object_server();
         let service = Self {
             collections: Default::default(),
             manager: Default::default(),
             connection: connection.clone(),
         };
-        let collections = service.fetch_collections().await;
 
-        object_server
+        connection
+            .object_server()
             .at(oo7::dbus::api::Service::PATH.as_deref().unwrap(), service)
             .await?;
 
-        for collection in collections {
-            object_server
-                .at(collection.path().clone(), collection)
+        Ok(connection)
+    }
+
+    pub async fn dispatch_collections(
```

Comment:
This should be called from Service::run and split it into two functions, one that takes care of handling session collection & the other for login collection

Reply:
With that you wouldn't have to deal with None secret at all and make the code much cleaner

Reply:
Done, thanks!